var/spack/repos/builtin/packages/magma/package.py

```diff
@@ -131,6 +131,8 @@ def cmake_args(self):
             sep = "" if "@:2.2.0" in spec else "_"
             capabilities = " ".join("sm{0}{1}".format(sep, i) for i in cuda_arch)
             options.append(define("GPU_TARGET", capabilities))
+            archs = ";".join("%s" % i for i in cuda_arch)
+            options.append(define("CMAKE_CUDA_ARCHITECTURES", archs))
```

Comment:
By the way, @stomov and @luszczek, this is something you may want to fix in MAGMA's CMake file: the value(s) set via `GPU_TARGET` are NOT actually used -- the processed arch flags are copied to the variable `CUDA_ARCHITECTURES` (see https://bitbucket.org/icl/magma/src/965af99469cf9b5fdd584b38a4d2911d21cb026c/CMakeLists.txt#lines-299) instead of `CMAKE_CUDA_ARCHITECTURES`.